File: Bridge/cpp/ZuluTimeImp.cpp
#include "ZuluTimeImp.h"

ZuluTimeImp::ZuluTimeImp(int hr, int min, int zone): TimeImp(hr, min) {
	if (zone == 5){
	  strcpy(zone_, " Eastern Standard Time");
	}else if (zone == 6){
	  strcpy(zone_, " Central Standard Time");
	 }
}

/* virtual */
void ZuluTimeImp::tell() {
	cout << "time is " << setw(2) << setfill('0') << hr_ << min_ << zone_ << endl;
}

File: Bridge/cpp/java/src/dp/bridge/time/ZuluTimeImp.java
package dp.bridge.time;

public class ZuluTimeImp extends TimeImp {
	protected String zoneStr;
	
	public ZuluTimeImp(int hr, int min, int zone){
		super(hr, min);
		if(zone == 5){
			this.zoneStr = " Eastern Standard Time";
		} else if(zone == 6){
			this.zoneStr = " Central Standard Time";
		}
	}
	
	public void tell(){
		System.out.println("time is " + this.hr + ":" + this.min + this.zoneStr);
	}
}